lib/SourceMapDevToolPlugin.js

```diff
@@ -129,25 +130,26 @@ SourceMapDevToolPlugin.prototype.apply = function(compiler) {
 					currentSourceMappingURLComment = currentSourceMappingURLComment.replace(/^\n\/\/(.*)$/, "\n/*$1*/");
 				}
 				if(sourceMapFilename) {
-					var filename = file,
-						query = "";
-					var idx = filename.indexOf("?");
-					if(idx >= 0) {
-						query = filename.substr(idx);
-						filename = filename.substr(0, idx);
+					var filename = splitQuery(file);
+					var contentHash;
+					if(sourceMapFilename.indexOf("[hash]")) {
```

Comment:
[hash] is usually used for the compilation hash. you should keep this behavior and add [contenthash] instead. best do this by adding a content property to this.getPath (see templatedpathplugin).